### PR TITLE
Update for Actions Windows MSYS2 install

### PR DESCRIPTION
**Pull request by MSP-Greg**
1. Use 'Git for Windows' cert.pem file
2. Add MSYS2 to path for mswin (after msvc additions)
3. Rearrange code for cleanup when MSYS2 is added to Window image

The comment below precedes code that can be removed when MSYS2 is added.
```
// Remove when Actions Windows image contains MSYS2 install
```

### CODE REVIEW

File: windows.js [hunk 1]
```diff
@@ -58,36 +62,37 @@ function latestHostedRuby() {
...
-    fs.copyFileSync(hostedCert, cert)
+    fs.copyFileSync(certFile, cert)
   }
-
-  return addVCVARSEnv()
+  let pathAry = addVCVARSEnv()
+  // add MSYS2 paths for misc gnu utilities like bison and ragel
+  pathAry.push(`${msys2}\\mingw64\\bin`)
+  pathAry.push(`${msys2}\\usr\\bin`)
+  return pathAry
```
Comment (eregon) [inline windows.js]: Could you use an array literal with `...` to splat addVCVARSEnv()?
That would be closer to the code above in `setupMingw` and avoid mutating the array in place.

File: windows.js [hunk 2]
```diff
@@ -8,6 +8,12 @@ const exec = require('@actions/exec')
 const tc = require('@actions/tool-cache')
 const rubyInstallerVersions = require('./windows-versions').versions
 
+// needed for 2.2, 2.3, and mswin, cert file used by Git for Windows
+const certFile = 'C:\\Program Files\\Git\\mingw64\\ssl\\cert.pem'
```
Comment (eregon) [inline windows.js]: I hope this location will remain available, it sounds likely enough for now.

Comment (MSP-Greg) [reply windows.js]: Git for Windows has always been there.  It's the git that's used by all the actions and any Windows devs using git.  There are also GitHub/MSFT staff involved with it.  IOW, its location is probably more locked down than the locations of the toolcache Rubies...

File: windows.js (hunk 1 above)
Comment (MSP-Greg) [reply windows.js]: I noticed that push allows multiple arguments, so I changed to that?   VCVARSEnv is defined by the bat file, so we have to 'add' two arrays, but this is just adding two items...

Comment (eregon) [reply windows.js]: I meant https://github.com/ruby/setup-ruby/commit/ae39a8d7d26f4a326f81694e5e99ed009d312fdf actually

**Comment by eregon** [comment]
Released as https://github.com/ruby/setup-ruby/releases/tag/v1.24.0

**Pull request merged**